src/Pages/UserProfile.jsx

```diff
@@ -0,0 +1,75 @@
+import React, { useState, useEffect } from "react";
+import "../style/userprofile.css";
+import { Link } from "react-router-dom";
+import { ImCross } from "react-icons/im";
+import axios from "axios";
+import { toast, ToastContainer } from "react-toastify";
+
+const UserProfile = () => {
+  const [userProfile, setUserProfile] = useState({
+    fullName: "",
+    email: "",
+    mobileNumber: "",
+    username: "",
+  });
+  const apiurl = process.env.REACT_APP_API_URL;
+  useEffect(() => {
+    const userId = localStorage.getItem("userId");
+    const fetchUserProfile = async () => {
+      try {
+        const response = await axios.get(apiurl + "/api/users/" + userId, {
+          headers: {
+            'ngrok-skip-browser-warning': '111'
+          }
+         });
+        setUserProfile(response.data);
+      } catch (error) {
+        console.log(error);
+        toast.error("Error!", { position: "top-center" });
+      }
```

Comment:
API calls should be placed in seperate folder/file and the method should be called when needed

Reply:
Done